### PR TITLE
Add python-api-examples: speech-recognition-from-microphone.py

### PR DESCRIPTION
**Pull request by yfyeung**

### CODE REVIEW

File: python-api-examples/speech-recognition-from-microphone.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        sample_rate=16000,
+        feature_dim=80,
+    )
+    return recognizer
+
+
+def main():
+    print("Started! Please speak")
+    recognizer = create_recognizer()
+    sample_rate = recognizer.sample_rate
```
Comment (csukuangfj) [inline python-api-examples/speech-recognition-from-microphone.py]: Please use 16000 directly.

recognizer in sherpa-onnx does not have the attribute `sample_rate`.

You can find the details of OnlineRecognizer at
https://github.com/k2-fsa/sherpa-onnx/blob/master/sherpa-onnx/python/sherpa_onnx/online_recognizer.py#L17

File: python-api-examples/speech-recognition-from-microphone.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    samples_per_read = int(0.1 * sample_rate)  # 0.1 second = 100 ms
+    last_result = ""
+    with sd.InputStream(channels=1, dtype="float32", samplerate=sample_rate) as s:
+        while True:
+            samples, _ = s.read(samples_per_read)  # a blocking read
+            samples = samples.reshape(-1)
+            recognizer.accept_waveform(sample_rate, samples)
```
Comment (csukuangfj) [inline python-api-examples/speech-recognition-from-microphone.py]: please follow
https://github.com/k2-fsa/sherpa-onnx/blob/063c68c91257ec5cf528f94e0b28eca095f8e1fd/python-api-examples/decode-file.py#L48-L50

In `sherpa-onnx`, we have a `stream` per utterance, which is used to support batch processing.

File: python-api-examples/speech-recognition-from-microphone.py (hunk 1 above)
Comment (yfyeung) [reply python-api-examples/speech-recognition-from-microphone.py]: OK, I will fix it.

File: python-api-examples/speech-recognition-from-microphone.py (hunk 2 above)
Comment (yfyeung) [reply python-api-examples/speech-recognition-from-microphone.py]: OK, I will fix it.

**Comment by csukuangfj** [comment]
Thanks!

**Pull request merged**